addons/l10n_in_ewaybill_stock/tools/ewaybill_api.py

```diff
@@ -0,0 +1,207 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import logging
+import contextlib
+
+from datetime import timedelta
+from markupsafe import Markup
+
+from odoo import fields, _
+from odoo.addons.iap import jsonrpc
+from odoo.exceptions import AccessError
+from odoo.addons.l10n_in_edi.models.account_edi_format import DEFAULT_IAP_ENDPOINT, DEFAULT_IAP_TEST_ENDPOINT
+from odoo.addons.l10n_in_edi_ewaybill.models.error_codes import ERROR_CODES
+
+
+_logger = logging.getLogger(__name__)
+
+
+class EWayBillError(Exception):
+
+    def __init__(self, response):
+        self.error_json = self._set_missing_error_message(response)
+        self.error_json.setdefault('odoo_warning', [])
+        self.error_codes = self.get_error_codes()
+        super().__init__(response)
+
+    def _set_missing_error_message(self, response):
+        for error in response.get('error', []):
+            if error.get('code') and not error.get('message'):
+                error['message'] = self._find_missing_error_message(error.get('code'))
+        return response
+
+    @staticmethod
+    def _find_missing_error_message(code):
+        error_message = ERROR_CODES.get(code)
+        return error_message or _("We don't know the error message for this error code. Please contact support.")
+
+    def get_all_error_message(self):
+        return Markup("<br/>").join(
+            [Markup("[%s] %s") % (e.get("code"), e.get("message")) for e in self.error_json.get('error')]
```

Comment:
```suggestion
            "[%s] %s" % (e.get("code"), e.get("message")) for e in self.error_json.get('error')]
```
As [] is text and we have Markup from the join, it can work this way.